docs/advanced/scopes-loadcontexts.rst

```diff
@@ -0,0 +1,75 @@
+AssemblyLoadContext and Lifetime Scopes
+=======================================
+
+In .NET Core and .NET 5+, the `the AssemblyLoadContext was introduced <https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext>`_.
+
+This allowed developers to dynamically load and unload assemblies from their application, and is very useful for developers writing applications with plugin-based architectures.
```

Comment:
```suggestion
In .NET Core `the AssemblyLoadContext was introduced <https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext>`_, allowing developers to dynamically load and unload assemblies from their application. This is very useful for developers writing applications with plugin-based architectures.
```

I don't think there's a .NET 5+ that isn't effectively .NET Core.

No action right now, but [this kinda ties into the issue we have where we need more docs about plugins and assembly loading](https://github.com/autofac/Documentation/issues/101). Again, nothing to do right now, just "filing for future reference."